types/react/jsx-dev-runtime.d.ts

```diff
@@ -1,2 +1,34 @@
-// Expose `JSX` namespace in `global` namespace
-import './';
+import { ElementType, Fragment, Key, ReactElement } from '.';
+
+export { Fragment };
+
+export interface JSXSource {
+    /**
+     * The source file where the element originates from.
+     */
+    fileName?: string;
+
+    /**
+     * The line number where the element was created.
+     */
+    lineNumber?: number;
+
+    /**
+     * The column number where the element was created.
+     */
+    columnNumber?: number;
+}
+
+/**
+ * Create a React element.
+ *
+ * You should not use this function directly. Use JSX and a transpiler instead.
+ */
+export function jsxDEV(
+    type: ElementType,
```

Comment:
I actually do want to use these functions' types to typecheck JSX in TS itself, so there would be value in making them more accurate, this way the types would be better when such checking is enabled. Ideally, at least as good as JSX tags are checked today.

Reply:
This is unrelated to the type checking of JSX syntax. JSX expressions for the automatic runtime are checked using the global JSX namespace. See the [TypeScript handbook on JSX](https://www.typescriptlang.org/docs/handbook/jsx.html). These functions do get used at runtime, so I understand the confusion.

Reply:
@weswigham Can you share a bit more information how that will look? Specifically with regard to https://github.com/microsoft/TypeScript/issues/14729#issuecomment-1429414701?

In the past, performance concerns were raised. React's JSX factories were never really tested for use in JSX type-checking. To avoid a chicken-egg problem (JSX factories being poorly typed vs TS not consulting JSX factories because poor types), I'd be nice if you could share TS canaries once ready so that I can check how to best type the JSX factories out.